### DISCUSSION TITLE
Welcome to Program-Trading-Based-on-Webull Discussions!

### DISCUSSION THREAD

**Discussion by LukeWang01** [lookatwallstreet/Program-Trading-Based-on-Webull#7] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by storm-99** [comment]
Interested to learn from your program trading, but have problem starting the app.
![image](https://github.com/LukeWang01/Program-Trading-Based-on-Webull/assets/46198822/cb08ad02-e5df-4839-8ce7-eda6a874b3d3)
how to solve this to get started?

**Reply by storm-99** [reply]
Redo from scratch, cloned your repo, same error:
ModuleNotFoundError: No module named 'env._secrete'
![image](https://github.com/LukeWang01/Program-Trading-Based-on-Webull/assets/46198822/92c64325-c7e4-49db-8bcc-21413ab8bdd1)

**Reply by Cfomodz** [reply]
Same problem.

Install python-dotenv: If not already installed, you can add it to your project by running:
bash
Copy code
pip install python-dotenv
Create a .env File: In your project root, create a file named .env and add your Discord credentials like this:
plaintext
Copy code
DISCORD_NOTIFY_TOKEN=your_discord_bot_token
CHANNEL_ID_GENERAL=your_channel_id
Modify the Import in Your Code: Replace the current import statement with code that uses dotenv to load the environment variables. For example:
python
Copy code
from dotenv import load_dotenv
import os

load_dotenv()  # This loads the .env file at the project root

discord_notify_Token = os.getenv('DISCORD_NOTIFY_TOKEN')
channel_id_general = os.getenv('CHANNEL_ID_GENERAL')
Adjust Any References: If other parts of your codebase reference discord_notify_Token and channel_id_general, make sure they are compatible with how you're now loading these variables.

**Comment by LukeWang01** [comment]
@storm-99 Hi, this is because I didn't upload my env files which contain my password and token. Please refer to @Cfomodz's solution. I just merged his PR. Thank you.

**Comment by LukeWang01** [comment]
Hi, for all the programming traders. I moved to MooMoo API to build a trading bot: https://github.com/LukeWang01/TradingBot-Based-on-MooMoo

There is an official secured API from MooMoo, which is also a 0-fee broker.

Fast and Safe.

My strategy has been running and testing for months and it works well. If you have any questions, please contact me.
